Signal/src/ViewControllers/AppSettingsViewController.m

```diff
@@ -238,6 +211,90 @@ - (void)updateTableContents
     self.contents = contents;
 }
 
+- (UITableViewCell *)profileHeaderCell
+{
+    UITableViewCell *cell = [UITableViewCell new];
+    cell.preservesSuperviewLayoutMargins = YES;
+    cell.contentView.preservesSuperviewLayoutMargins = YES;
+    cell.selectionStyle = UITableViewCellSelectionStyleNone;
+
+    const NSUInteger kAvatarSize = 68;
+    // TODO: Replace this icon.
+    UIImage *_Nullable localProfileAvatarImage = [OWSProfileManager.sharedManager localProfileAvatarImage];
+    UIImage *avatarImage = (localProfileAvatarImage
+            ?: [[UIImage imageNamed:@"profile_avatar_default"]
+                   imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]);
+    OWSAssert(avatarImage);
+
+    AvatarImageView *avatarView = [[AvatarImageView alloc] initWithImage:avatarImage];
+    if (!localProfileAvatarImage) {
+        avatarView.tintColor = [UIColor colorWithRGBHex:0x888888];
+    }
+    [cell.contentView addSubview:avatarView];
+    [avatarView autoVCenterInSuperview];
+    [avatarView autoPinLeadingToSuperView];
+    [avatarView autoSetDimension:ALDimensionWidth toSize:kAvatarSize];
+    [avatarView autoSetDimension:ALDimensionHeight toSize:kAvatarSize];
+
+    if (!localProfileAvatarImage) {
+        UIImage *cameraImage = [UIImage imageNamed:@"settings-avatar-camera"];
+        cameraImage = [cameraImage imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+        UIImageView *cameraImageView = [[UIImageView alloc] initWithImage:cameraImage];
+        cameraImageView.tintColor = [UIColor ows_materialBlueColor];
+        [cell.contentView addSubview:cameraImageView];
+        [cameraImageView autoPinTrailingToView:avatarView];
+        [cameraImageView autoPinEdge:ALEdgeBottom toEdge:ALEdgeBottom ofView:avatarView];
+    }
+
+    UIView *threadNameView = [UIView containerView];
```

Comment:
We are now showing a "avatar + name + optional subtitle" header in multiple places: profile view, conversation settings, new group, update group.  This code is cribbed from one of those other locations.  I'd like to eventually pull out a widget that we can hopefully use in all of those locations, so my intent was to try to keep the code as similar as possible for now. Reasonable or should I rename?